sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/EncodingPerFileTest.java

```diff
@@ -51,40 +50,43 @@ public void should_treat_as_match_when_roslyn_entry_missing_for_file() throws IO
     Charset roslynCharset = StandardCharsets.UTF_8;
     Charset sqCharset = StandardCharsets.UTF_16;
 
-    HashMap<URI, Charset> encodingPerUri = new HashMap<>();
-    encodingPerUri.put(fileUri, roslynCharset);
-    assertEncodingMatch(encodingPerUri, URI.create("dummy"), sqCharset, true);
+    assertEncodingMatch(roslynCharset, URI.create("dummy"), sqCharset, true);
   }
 
   @Test
   public void should_treat_as_mismatch_when_roslyn_encoding_missing() throws IOException {
-    HashMap<URI, Charset> encodingPerUri = new HashMap<>();
-    encodingPerUri.put(fileUri, null);
-    assertEncodingMatch(encodingPerUri, fileUri, null, false);
+    assertEncodingMatch(null, fileUri, null, false);
   }
 
   @Test
   public void should_treat_as_mismatch_when_roslyn_utf8_and_sq_utf16() throws IOException {
     Charset roslynCharset = StandardCharsets.UTF_8;
     Charset sqCharset = StandardCharsets.UTF_16;
 
-    HashMap<URI, Charset> encodingPerUri = new HashMap<>();
-    encodingPerUri.put(fileUri, roslynCharset);
-    assertEncodingMatch(encodingPerUri, fileUri, sqCharset, false);
+    assertEncodingMatch(roslynCharset, fileUri, sqCharset, false);
```

Comment:
hmm reading this test it's very hard to understand the difference between this and `should_treat_as_match_when_roslyn_entry_missing_for_file`

why is `URI.create("dummy")` different from `fileUri` which is `temp.newFile().toURI();` ? 

because the first does not exist on disk and the `fileUri` does exist on disk?

Reply:
I think the names describe their contents well. 
`entry_missing_for_file` - entire entry for the file itself is missing in the list. 
`encoding_missing` - the entry for the file exists in the list, but contains charset NULL